### PR TITLE
(SIMP-8382) Confine tmp_mounts on directories

### PR DESCRIPTION
**Pull request by trevor-vaughan**
SIMP-8382 #close

### CODE REVIEW

File: lib/facter/tmp_mounts.rb [hunk 1]
```diff
@@ -22,6 +22,8 @@
 simplib__tmp_mount_target_dirs.each do |dir|
   Facter.add("tmp_mount#{dir.gsub('/','_')}") do
     setcode do
+      confine { File.directory?(dir) }
```
Comment (lnemsick-simp) [inline lib/facter/tmp_mounts.rb:25]: The original code had `confine :kernel => :linux`.  Is there a reason to switch to this directory confine?

Comment (lnemsick-simp) [reply lib/facter/tmp_mounts.rb:25]: The original code only created the fact if the directory existed and then confined based on kernel.  Will just confining based on directory be sufficient?

Comment (trevor-vaughan) [reply lib/facter/tmp_mounts.rb]: @lnemsick-simp Yep. Windows certainly won't have that directory.

**Comment by trevor-vaughan** [comment]
Adding an acceptance test for Windows to make sure we don't blow it up unexpectedly again

**Review by lnemsick-simp — APPROVED** [review]

**Pull request merged**